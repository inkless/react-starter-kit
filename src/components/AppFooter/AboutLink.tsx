import React from 'react';
import styled, { StyledProps } from 'styled-components';
import { NavLink } from 'react-router-dom';

function AboutLink(props: StyledProps<{}>) {
  return (
    <NavLink {...props} to="/about">
      About
    </NavLink>
  );
}

export const StyledAboutLink = styled(AboutLink)`
  text-decoration: none;
  color: ${p => p.theme.colors.common.white};

  &:hover {
    text-decoration: underline;
  }
`;

export default StyledAboutLink;
