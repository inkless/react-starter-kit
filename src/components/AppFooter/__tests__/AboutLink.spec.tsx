import React from 'react';
import { shallow } from 'enzyme';

import AboutLink from '../AboutLink';

describe('Component AboutLink', () => {
  it('should render AboutLink correctly', () => {
    const component = shallow(<AboutLink />);

    expect(component).toMatchSnapshot();
  });
});
