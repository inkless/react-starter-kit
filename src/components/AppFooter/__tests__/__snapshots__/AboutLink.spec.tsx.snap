// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component AboutLink should render AboutLink correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  text-decoration: none;
  color: ",
          [Function],
          ";

  &:hover {
    text-decoration: underline;
  }
",
        ],
      },
      "displayName": "Styled(AboutLink)",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
